lib/shred.ts

```diff
@@ -73,19 +84,20 @@ exports.shredRecord = function(schema, record, buffer) {
       }
     }
 
-    [...recordShredded[field.path].distinct_values].forEach(value => buffer.columnData[field.path].distinct_values.add(value));
+    [...recordShredded[path].distinct_values].forEach(value => buffer.columnData[path].distinct_values.add(value));
 
-    buffer.columnData[field.path].count += recordShredded[field.path].count;
+    buffer.columnData[path].count += recordShredded[path].count;
   }
 };
 
-function shredRecordInternal(fields, record, data, rlvl, dlvl) {
+function shredRecordInternal(fields: Record<string, ParquetField>, record: Record<string, object[]> | null, data: Record<string, ParquetData>, rlvl: number, dlvl: number) {
   for (let fieldName in fields) {
     const field = fields[fieldName];
     const fieldType = field.originalType || field.primitiveType;
+    const path = field.path as string
 
     // fetch values
-    let values = [];
+    let values: Array<object> = [];
```

Comment:
Based on how they're used values is an Array of some object but is also used to be passed back in as the`record` parameter. However the record parameter has to be some type of key,value structure to use lines such as `record[fieldName]`. Currently struggling to figure out a fix